test/extended/util/framework.go

```diff
@@ -1691,12 +1721,16 @@ type podExecutor struct {
 
 // NewPodExecutor returns an executor capable of running commands in a Pod.
 func NewPodExecutor(oc *CLI, name, image string) (*podExecutor, error) {
-	out, err := oc.Run("run").Args(name, "--labels", "name="+name, "--image", image, "--restart", "Never", "--command", "--", "/bin/bash", "-c", "sleep infinity").Output()
+	overrides := fmt.Sprintf(`{"spec":{"containers":[{"name":"%s","terminationMessagePolicy":"FallbackToLogsOnError"}]}}`, name)
+	out, err := oc.Run("run").Args(name, "--labels", "name="+name, "--image", image, "--restart", "Never", "--overrides", overrides, "--override-type", "strategic", "--command", "--", "/bin/bash", "-c", "sleep infinity").Output()
```

Comment:
Can you either use https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/pod/create.go#L80 or create something like it in code so that we can easily manipulate the pod as need for hetero and other purposes.